File: 6KYU/Decode the Morse Code/solution.cs
using System.Linq;
using System;
class MorseCodeDecoder
{
	public static string Decode(string morseCode)
		=> string.Join(" ", morseCode.TrimStart().TrimEnd().Split("   ").Select(c => String.Concat(c.Split().Select(m => MorseCode.Get(m)))));
}

File: 6KYU/Decode the Morse Code/solution.go
package kata

import "strings"

func DecodeMorse(morseCode string) string {
	trimmed := strings.Trim(morseCode, " ")
	words := strings.Split(trimmed, "   ")
	res := []string{}
	for _, word := range words {
		letters := strings.Split(word, " ")
		decodedWord := ""
		for _, letter := range letters {
			decodedWord += MORSE_CODE[letter]
		}
		res = append(res, decodedWord)
	}
	return strings.Join(res, " ")
}
